zip-0226.rst

```diff
@@ -1,8 +1,305 @@
+  License: MIT
   Discussions-To: <https://github.com/zcash/zips/issues/618>
+
+
+Terminology
+===========
+
+The key word "MUST" in this document is to be interpreted as described in RFC 2119 [#RFC2119]_.
+
+The term "network upgrade" in this document is to be interpreted as described in ZIP 200 [#zip-0200]_.
+
+The term "Orchard" in this document is to be interpreted as described in ZIP 224 [#zip-0224]_.
+
+We define the following additional terms:
+
+- Asset: A type of note that can be transferred on the Zcash block chain, identified by the :math:`\mathsf{AssetId}` parameter.
+
+  - ZEC is the default (and currently the only defined) Asset for the Zcash mainnet.
+  - We use the term "custom Asset" to refer to any Asset other than ZEC.
+
+- Zcash Shielded Asset: an Asset with issuance defined on the Zcash block chain, and, for now, belonging to the Orchard shielded pool.
+- Wrapped Asset: a Zcash Shielded Asset with native issuance defined outside the Zcash block chain.
+- Split Input: an Action input identifying a Zcash Shielded Asset note, used to ensure that the output note of that Action is of a validly issued :math:`\mathsf{AssetId}` when there is no corresponding real input note to spend. The Action does not spend the note identified by the Split Input.
+- Split Action: a Zcash Shielded Asset Action that contains a Split Input.
+
+Abstract
+========
+
+ZIP 226 and ZIP 227 propose in conjunction the Zcash Shielded Assets (ZSA) protocol, which is an extension of the
+Orchard protocol that enables the creation, transfer, and burn of custom Assets on the Zcash chain. The creation of such Assets is defined
+in ZIP 227 [#zip-0227]_. The transfer and burn of such Assets is defined in ZIP 226 [#zip-0226]_.
+
+Motivation
+==========
+
+The Orchard protocol deployed in NU5 does not support custom Assets. Enabling multi-asset support on the Zcash chain will open the door for a host of applications, and enhance the ecosystem with application developers and Asset custody institutions for issuance and bridging purposes.
+
+Overview
+========
+In order to be able to represent different Asset identifiers in Orchard, we need to define a new data field that uniquely represents the identifier of the Asset in question, which we call :math:`\mathsf{AssetId}`.
+
+The Asset identifier will be used to enforce the balance of an Action Description [#protocol-actions]_ is preserved across Assets (see the Orchard Binding Signature [#protocol-binding]_), and by extension the balance of an Orchard transaction. That is, the sum of all the :math:`\mathsf{value^{net}}` from each Action Description, computed as :math:`\mathsf{value^{old}-value^{new}}`, must be balanced **only with respect to the same Asset identifier**. This is specially important since we will allow different Action Descriptions to transfer notes of different Asset identifiers, where the overall balance is checked without revealing which Assets (or how many different identifiers) are being transferred.
```

Comment:
```suggestion
The Asset identifier will be used to enforce that the balance of an Action Description [#protocol-actions]_ is preserved across Assets (see the Orchard Binding Signature [#protocol-binding]_), and by extension the balance of an Orchard transaction. That is, the sum of all the :math:`\mathsf{value^{net}}` from each Action Description, computed as :math:`\mathsf{value^{old}-value^{new}}`, must be balanced **only with respect to the same Asset identifier**. This is specially important since we will allow different Action Descriptions to transfer notes of different Asset identifiers, where the overall balance is checked without revealing which Assets (or how many different identifiers) are being transferred.
```

Reply:
Done.